File: src/main/java/com/elixir/mars/repository/models/isa/Materials.java
package com.elixir.mars.repository.models.isa;

import java.util.ArrayList;
import lombok.Data;
import lombok.experimental.FieldNameConstants;

@Data
@FieldNameConstants
public class Materials {
  public ArrayList<Source> sources;
  public ArrayList<Sample> samples;
  public ArrayList<OtherMaterial> otherMaterials;
}

File: python/mars_repository/models/isa/materials.py
from __future__ import annotations
from typing import Any, List, TypedDict
from .source import Source
from .sample import Sample
from .other_material import OtherMaterial

class Materials(TypedDict, total=False):
    sources: List[Source]
    samples: List[Sample]
    other_materials: List[OtherMaterial]
